src/main/java/dev/slimevr/vr/processor/skeleton/SimpleSkeleton.java

```diff
@@ -704,85 +715,171 @@ protected void updateLocalTransforms() {
 		trackerRightFootNode.localTransform.setRotation(rotBuf2);
 
-		if (rightFootTracker != null) {
-			rightFootTracker.getRotation(rotBuf2);
-			rightAnkleNode.localTransform.setRotation(rotBuf2);
-			rightFootNode.localTransform.setRotation(rotBuf2);
-			trackerRightFootNode.localTransform.setRotation(rotBuf2);
+		// Extended spine
+		if (extendedSpineModel && hasSpineTracker) {
+			if (
+				(chestTracker != null && (waistTracker == null || hipTracker == null))
+					|| (waistTracker != null && hipTracker == null)
+			) {
+				// Tries to guess missing lower spine trackers by interpolating
+				// rotations
+				if (waistTracker == null) {
+					if (hipTracker != null) {
+						// Calculates waist from chest + hip
+						chestTracker.getRotation(rotBuf1);
+						hipTracker.getRotation(rotBuf2);
+
+						rotBuf1.slerpLocal(rotBuf2, waistChestHipFactor);
+						chestNode.localTransform.setRotation(rotBuf1);
+					} else if (hasKneeTrackers) {
+						// Calculates waist from chest + pelvis
+						leftHipNode.localTransform.getRotation(rotBuf1);
+						rightHipNode.localTransform.getRotation(rotBuf2);
+						chestTracker.getRotation(rotBuf3);
+
+						// Get the rotation relative to where we expect the
+						// upper legs to be
+						rotBuf3.mult(rotBuf4.fromAngles(FastMath.HALF_PI, 0, 0), rotBuf4);
+						if (rotBuf4.dot(rotBuf1) < 0.0f) {
+							rotBuf1.negateLocal();
+						}
+						if (rotBuf4.dot(rotBuf2) < 0.0f) {
+							rotBuf2.negateLocal();
+						}
+
+						rotBuf1.nlerp(rotBuf2, 0.5f);
+						rotBuf3.pureSlerpLocal(rotBuf1, waistChestPelvisFactor);
+						chestNode.localTransform.setRotation(rotBuf3);
+					}
+				}
+				if (hipTracker == null && hasKneeTrackers) {
+					// Calculates hip from (chest or waist) + pelvis
+					leftHipNode.localTransform.getRotation(rotBuf1);
+					rightHipNode.localTransform.getRotation(rotBuf2);
+					getFirstAvailableTracker(waistTracker, chestTracker, null).getRotation(rotBuf3);
+
+					// Get the rotation relative to where we expect the upper
+					// legs to be
+					rotBuf3.mult(rotBuf4.fromAngles(FastMath.HALF_PI, 0, 0), rotBuf4);
+					if (rotBuf4.dot(rotBuf1) < 0.0f) {
+						rotBuf1.negateLocal();
+					}
+					if (rotBuf4.dot(rotBuf2) < 0.0f) {
+						rotBuf2.negateLocal();
+					}
+
+					rotBuf1.nlerp(rotBuf2, 0.5f);
+					rotBuf3.pureSlerpLocal(rotBuf1, hipSpinePelvisFactor);
+					waistNode.localTransform.setRotation(rotBuf3);
+				}
+			}
 		}
 
-		if (extendedPelvisModel && hasKneeTracker) {
+		// Extended pelvis
+		if (extendedPelvisModel && hasKneeTrackers) {
 			// Average pelvis between two legs
 			leftHipNode.localTransform.getRotation(rotBuf1);
 			rightHipNode.localTransform.getRotation(rotBuf2);
 			rotBuf2.nlerp(rotBuf1, 0.5f);
-			chestNode.localTransform.getRotation(rotBuf1);
-			rotBuf2.nlerp(rotBuf1, FastMath.ONE_THIRD);
+			waistNode.localTransform.getRotation(rotBuf1);
+
+			rotBuf2.slerpLocal(rotBuf1, FastMath.ONE_THIRD);
```

Comment:
Maybe the `FastMath.ONE_THIRD` value could be configurable? Would that make sense for this?

Reply:
yes indeed